csharp/private/wrappers/dotnet.cc

```diff
@@ -50,14 +63,34 @@ int main(int argc, char** argv) {
       evprintf("DOTNET_CLI_TELEMETRY_OPTOUT", "1"),  // disable telemetry
   };
 
+  auto extra_count =
+    !!(std::strlen(kArgv1) != 0) +
+    !!(std::strlen(kArgv2) != 0);
+
+  auto dotnet_argv = new char*[extra_count + argc + 1];
+
   // dotnet wants this to either be dotnet or dotnet.exe but doesn't have a
   // preference otherwise.
-  auto dotnet_argv = new char*[argc + 1];
-  dotnet_argv[0] = (char*)"dotnet";
+  dotnet_argv[0] = const_cast<char*>("dotnet");
+
+  // Make sure to hold a reference to these std::string.
+  auto argv1 = runfiles->Rlocation(kArgv1);
+  auto argv2 = runfiles->Rlocation(kArgv2);
+
+  if (std::strlen(kArgv1) != 0) {
+    dotnet_argv[1] = const_cast<char*>(argv1.c_str());
+  }
+
+  if (std::strlen(kArgv2) != 0) {
+    dotnet_argv[2] = const_cast<char*>(argv2.c_str());
+  }
+
+  // Copy the rest of our arguments in
   for (int i = 1; i < argc; i++) {
-    dotnet_argv[i] = argv[i];
+    dotnet_argv[i + extra_count] = argv[i];
```

Comment:
nit: prefer extra_count stayed consistently at the front

Reply:
Ah yeah good catch.